zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -220,7 +227,15 @@ protected ZkClient(IZkConnection zkConnection, int connectionTimeout, long opera
       }
       listeners.add(listener);
     }
-    return watchForChilds(path);
+
+    List<String> children = watchForChilds(path, skipWatchingNodeNotExist);
+    if (children == null && skipWatchingNodeNotExist) {
+      unsubscribeChildChanges(path, listener);
+      LOG.info("watchForChilds failed to install no-existing watch and add listener. Path:" + path);
```

Comment:
This api is run frequently. A good idea to use parameterize logging to save some string concatenation overhead.

Reply:
fixed.